### PR TITLE
[SWE-73] Add Profile Picture

### PR DESCRIPTION
**Pull request by datderek**
Task ID: [SWE-73](https://linear.app/swe-dem-boys/issue/SWE-73/add-profile-picture)
## Task Description
Allow users to upload a picture.
## Changes Made
- `EditProfileForm.tsx` now uploads the profile picture file to S3 buckets such that users can change their pfp
- `ProfileIcon.tsx` now retrieves the profile picture when provided with the userId
- `Navbar.tsx` and `Post.tsx` display the correct profile picture by providing the userId
## Notes
URI is NOT stored in the user vertex. All profile pictures are store in S3 buckets in with the following path: '{userId}/pfp'. Thus a request can be made for any pfp provided you have the userId.

**ALSO** the user profile pages still do not display the correct image. I don't how to make this work...? Currently we need the userId to retrieve their profile picture; however, we don't have an API yet to retrieve a userId based on their username? (i.e. when we visit a users profile page)

### CODE REVIEW

**Review by sammyoto — APPROVED** [review]
Code looks good to me, about not being able to load other profiles profile pictures I think that shouldn't be to hard for us to figure out. We can talk about it tomorrow at our meeting.

File: app/components/ClientUtilityFunctions.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { useRouter } from 'next/navigation';
  * 
  * @param isUser - true if current session user matches page, false otherwise
  */
-export function EditProfileAndFollowButton({ user, isUser }: { user: string, isUser: boolean }) {
+export function EditProfileAndFollowButton({ user, isUser }: { user: String, isUser: boolean }) {
```
Comment (jackfales) [inline app/components/ClientUtilityFunctions.tsx:9]: Maybe `isCurrentUser` or something similar? The assumption is that if you are viewing a page you are a user so the variable name doesn't describe much.

**Review by jackfales — APPROVED** [review]
Overall LGTM, one minor nit comment

File: app/components/ClientUtilityFunctions.tsx (hunk 1 above)
Comment (datderek) [reply app/components/ClientUtilityFunctions.tsx:9]: Agreed agreed.

**Review by jackfales — APPROVED** [review]

**Pull request merged**